src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<T>(lastNode, value, null);
+        if (firstNode == null) {
+            firstNode = node;
+            lastNode = node;
+            size++;
+            return;
+        }
+        if (firstNode.next == null) {
+            firstNode.next = node;
+        }
```

Comment:
This condition doesn't seem necessary. If you have only 1 node then both firstNode and lastNode point to same object so in this case lines 20 and 22 are just duplication each other
```suggestion
```